### PR TITLE
[ntuple] Fix reading of untyped collections

### PR DESCRIPTION
**Pull request by jblomer**
Along the way, this PR removes the `RFieldFuse` helper class. Instead, the RField class knows its field ID where available.  This simplifies connecting fields to a page storage and gives the process more flexibility. The PR also adds an `RRecordField` for untyped `struct`s.

@mxxo That should fix the problem you saw with NanoAOD printing.

### CODE REVIEW

**Comment by jblomer** [comment]
This PR should also help with #6700

File: tree/dataframe/src/RNTupleDS.cxx [hunk 1]
```diff
@@ -47,9 +47,10 @@ class RNTupleColumnReader final : public ROOT::Detail::RDF::RColumnReaderBase {
       const auto &descriptor = source.GetDescriptor();
       const auto fieldId = descriptor.FindFieldId(colName);
       const auto &fieldDescriptor = descriptor.GetFieldDescriptor(fieldId);
-      const auto typeName = fieldDescriptor.GetTypeName();
-      auto fieldBasePtr = Detail::RFieldBase::Create(fieldDescriptor.GetFieldName(), typeName).Unwrap();
-      Detail::RFieldFuse::ConnectRecursively(fieldId, source, *fieldBasePtr);
+      auto fieldBasePtr = fieldDescriptor.CreateField(descriptor);
+      fieldBasePtr->ConnectPageStorage(source);
+      for (auto &f : *fieldBasePtr)
```
Comment (pcanal) [inline tree/dataframe/src/RNTupleDS.cxx:52]: This seems to imply to ConnectPageStorage is not recursive but the loop below goes only one level 'deep'.  Is there only one level of nesting to the fields?

File: tree/ntuple/v7/inc/ROOT/RNTupleView.hxx [hunk 2]
```diff
@@ -164,7 +164,13 @@ private:
    RNTupleView(DescriptorId_t fieldId, Detail::RPageSource* pageSource)
      : fField(pageSource->GetDescriptor().GetFieldDescriptor(fieldId).GetFieldName()), fValue(fField.GenerateValue())
    {
-      Detail::RFieldFuse::ConnectRecursively(fieldId, *pageSource, fField);
+      fField.SetOnDiskId(fieldId);
+      fField.ConnectPageStorage(*pageSource);
```
Comment (pcanal) [inline tree/ntuple/v7/inc/ROOT/RNTupleView.hxx:168]: It is not clear (and/or I missed the description) why `SetOnDiskId` is called along side `ConnectPageStorage
` and why sometimes only `ConnectPageStorage` is called.  Could you clarify?

File: tree/dataframe/src/RNTupleDS.cxx (hunk 1 above)
Comment (jblomer) [reply tree/dataframe/src/RNTupleDS.cxx:52]: That's right, `ConnectPageStorage` is not recursive but the field iterator traverses the entire subtree so the net effect is the same than the previous `Detail::RFieldFuse::ConnectRecursively`.

File: tree/ntuple/v7/inc/ROOT/RNTupleView.hxx (hunk 2 above)
Comment (jblomer) [reply tree/ntuple/v7/inc/ROOT/RNTupleView.hxx:168]: The separation of `SetOnDiskId` and `ConnectPageStorage` allows for lazy/late binding. We can create fields and remember to which field ID on disk they refer to but only acquire actual I/O buffers when we really need the field.  That's handy for RDF, for example.  When an RNTuple view is created, however, the underlying field should immediately be available for reading, so we call both `SetOnDiskId` and `ConnectPageStorage` at the same time.

**Review by mxxo — APPROVED** [review]
Looks good to me! and I like that `RFieldFuse` was dethroned along the way. Having `_0` as a field name in the printout is maybe a little strange but I understand the motivation and this is a huge improvement.

**Pull request merged**